### PR TITLE
feat(snap): add startup-duration and startup-interval configure options

### PR DESCRIPTION
**Pull request by siggiskulason**
To change the default startup duration use:
sudo snap set edgex-device-mqtt startup-duration=n

To change the default startup interval use:
sudo snap set edgex-device-mqtt startup-interval=n

Signed-off-by: Siggi Skulason <siggi.skulason@canonical.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

The module go-mod-bootstrap implements a service dependency check on startup which ensures that all of the runtime dependencies of a particular service are met before the service transitions to active state.

The logic implements a basic retry loop with a maximum duration and sleep interval. If the dependent services are not available, the service sleeps for the defined interval (default: 1s) and then tries again up to a maximum duration (default: 60s). These values can be overridden via environment variables.

Issue Number:
N/A

## What is the new behavior?
This PR covers updating the device service snap to allow these variables to be set/queried via the configure hook.

To change the default startup duration use:
sudo snap set edgex-device-mqtt startup-duration=n

To change the default startup interval use:
sudo snap set edgex-device-mqtt startup-interval=n

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Are there any new imports or modules? If so, what are they used for and why?
N/A

## Are there any specific instructions or things that should be known prior to reviewing?
N/A

## Other information
N/A

### CODE REVIEW

**Review by lenny-intel — DISMISSED** [review]
LGTM

**Review by tonyespy — CHANGES_REQUESTED** [review]
Looks good, just the one inline comment about the README, and the commit message nit (i.e. configure hooks vs. configure options).

Also as an aside, it looks like this service is still using a version-script for the snap?

File: snap/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+This folder contains snap packaging for the EdgeX MQTT Device Service Snap
+
+The snap currently supports both `amd64` and `arm64` platforms.
+
+
+## Snap configuration
+
+Device services implement a service dependency check on startup which ensures that all of the runtime dependencies of a particular service are met before the service transitions to active state.
+
+The logic implements a basic retry loop with a maximum duration and sleep interval. If the dependent services are not available, the service sleeps for the defined interval (default: 1s) and then tries again up to a maximum duration (default: 60s). These values can be overridden with the following commands:
```
[the remark this answers is not in the document]
Comment (siggiskulason) [reply snap/README.md]: Added explanation

**Comment by siggiskulason** [comment]
Updated as per comments:
- [x] updated comment: Added N/A
- [x] README.md: Added info about why we are doing this
- [x] Configure options - not configure hooks. Changed commit message and PR title

**Comment by siggiskulason** [comment]
> Also as an aside, it looks like this service is still using a version-script for the snap?
It is, yes, I think this is the only device snap still doing so. I'll create a separate PR for that change

**Review by tonyespy — APPROVED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]

**Pull request merged**